### DISCUSSION TITLE
kallisto pseudoalignment

### DISCUSSION THREAD

**Discussion by laurel-nave-powers** [sr320/course-fish546-2021#125] [Q&A] [answered]
I am now in the quantifying expression stage, in which it should pseudoalign the reads. My code works but it did not align the reads. It says to check my input files and index, but I'm not sure what to be looking for with that. 
Notebook: https://github.com/fish546-2021/Laurel-genes/blob/main/code/04_kallisto.ipynb

**Comment by sr320** [comment]
Lets maybe step back a bit...

You have DNA reads from two flatfishes--- trying to align to ??

Normally kallisto is used for RNA-seq.. 

Is this the same workflow Calder used?

**Reply by laurel-nave-powers** [reply]
I thought that the index file was what it is trying to align to - the index file is from Danio rerio. 
Sam had said that even though kallisto is used for RNA-seq it could be used for other data in a pinch. This seemed like the fastest way to get an alignment which is why I'm trying it. Maybe that's why it's not aligning? Maybe I need to change some of the code?
No, Calder's workflow with the perl scripts was giving me way too many speedbumps that I was unable to work through quickly enough. So, I'm just trying to get something at this point.

**Reply by kubu4** [reply]
The suggestion to run kallisto was primarily as a consolation to get some sort of alignment data in time for to use for the class project. 

Calder's pipeline was becoming a complete time suck, consisting of mostly troubleshooting. Unfortunately, the pipeline consists mostly of Perl wrapper scripts, with minimal documentation.

Big picture suggestion was to use GATK instead for the project, as it's sort of the standard set of tools/pipelines for this kind of analysis. Plus, it's very well documented and has a plethora of examples for common analyses. However, there may not have been enough time to switch over to GATK and get something usable as a "product" for class purposes.

**Comment by sr320** [comment]
Can you provide urls to fastqs?

On Tue, Mar 9, 2021 at 1:59 PM laurel-nave-powers <notifications@github.com>
wrote:

> I thought that the index file was what it is trying to align to - the
> index file is from Danio rerio.
> Sam had said that even though kallisto is used for RNA-seq it could be
> used for other data in a pinch. This seemed like the fastest way to get an
> alignment which is why I'm trying it. Maybe that's why it's not aligning?
> Maybe I need to change some of the code?
> No, Calder's workflow with the perl scripts was giving me way too many
> speedbumps that I was unable to work through quickly enough. So, I'm just
> trying to get something at this point.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/sr320/course-fish546-2021/discussions/125#discussioncomment-457558>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABB4PN5WLQ5XRQQEZHVP4R3TC2D2RANCNFSM4Y4O253Q>
> .
>
--

**Reply by laurel-nave-powers** [reply]
My data isn't online, just on my computer. 
But here's links to fastqc results if that's helpful. 
file:///Users/laurelnave-powers/Desktop/GitHub/Laurel-genes/analyses/fastqc/HIPPLAT_NEF95054_S2_R1_fastqc.html
file:///Users/laurelnave-powers/Desktop/GitHub/Laurel-genes/analyses/fastqc/HIPPLAT_NEF95054_S2_R2_fastqc.html
file:///Users/laurelnave-powers/Desktop/GitHub/Laurel-genes/analyses/fastqc/MICPACI_UW157426_S7_R1_fastqc.html
file:///Users/laurelnave-powers/Desktop/GitHub/Laurel-genes/analyses/fastqc/MICPACI_UW157426_S7_R2_fastqc.html

**Comment by aspencoyle** [comment]
This might be the issue! `-o` should be a path to where your kallisto output is stored. Here's an example (literally running this right now)!

Working directory also has kallisto_hemat_index_v1.6.idx

Also ../kallisto_libraries_hemat_transcriptomev1.6/id334 doesn't need to exist prior to running kallisto, it makes the directory for me

Finally, the 2>> bit is me redirecting standard error to a file - yours should run just fine with or without that part (although if included you won't see any output on the screen)

```
# Quantify ID 334
kallisto quant \
-i kallisto_hemat_index_v1.6.idx \
-o ../kallisto_libraries_hemat_transcriptomev1.6/id334 \
../../data/libraries/334_R1_001.fastp-trim.202003181149.fq.gz \
../../data/libraries/334_R2_001.fastp-trim.202003181149.fq.gz \
2>> ../kallisto_libraries_hemat_transcriptomev1.6/std_errortracking.txt
```

**Comment by laurel-nave-powers** [comment, marked as answer]
Update: It finally finished running, and I think it worked! https://github.com/fish546-2021/Laurel-genes/blob/main/data/bowtie.ipynb 
Changing the file to be indexed from the .alt to the .sm seemed to do the trick.

**Reply by kubu4** [reply]
> Changing the file to be indexed from the .alt to the .sm seemed to do the trick.

I think I'm missing something here. It looks like you switched from using kallisto to using bowtie2 (which is great - this is the better approach; just wasn't sure how long this would take to run on your own computer); which isn't related to switching the reference file.

**Reply by sr320** [reply]
had wrong genome... we figured out in office ours.
